components/ee/payment-endpoint/src/github/subscription-reconciler.ts

```diff
@@ -219,27 +212,30 @@ export class GithubSubscriptionReconciler {
         }), 24 * 60 * 1000); // once a day
     }
 
-    public async reconcileEvent(user: User, evt: MarketplaceEventAll) {
+    public async reconcileEvent(user: User, evt: MarketplaceEventAll, runInTransaction?: (manager: EntityManager) => Promise<void>) {
         const userId = user.id;
-        await this.accountingDB.transaction(async db => {
-            const subscriptions = await db.findAllSubscriptionsForUser(userId);
-            const userPaidSubscriptions = subscriptions.filter(s => UserPaidSubscription.is(s) && s.paymentReference.startsWith("github:"));
-
-            const model = new SubscriptionModel(userId, userPaidSubscriptions);
-            const success = await this.subscriptionMapper.map(evt, model);
-            if (!success) {
-                log.debug({userId}, "subscription mapper did not succeed for GitHub market purchase event. See errors above.", { evt });
-                return;
-            }
+        const subscriptions = await this.accountingDB.findAllSubscriptionsForUser(userId);
```

Comment:
There is a slight drift here: The read ops are not run in the same transaction as the writes as before, but I guess you're aware of that.

Reply:
> There is a slight drift here: The read ops are not run in the same transaction as the writes as before, but I guess you're aware of that.

Yes. That would only have implications if:
 - we had concurrent writers
 - rely on transaction-scoped locks to resolve those
 As we don't, this does not matter.